Pass the test name as a parameter to the run_ncat() function in system_tests_tcp_adaptor.py

Currently on an ncat failure, we see the following error message - 
```
E           Process 4421 (name=ncat) error: process was unexpectedly still running
E           ncat localhost 26119 --ssl-trustfile /home/runner/work/skupper-router/skupper-router/skupper-router/tests/ssl_certs/ca-certificate.pem
E           /__w/skupper-router/skupper-router/skupper-router/build/tests/system_test.dir/tests/system_tests_tcp_adaptor_tls/TcpTlsAdaptor/test_70_half_closed/ncat-4.cmd
```
It would be better if the Process name had the actual test being run so you know which router the ncat command ran against. For example, it would be nice if the above error looked like this - 
```
Process 4421 (name=test_70_half_closed_INTA_INTA) error: process was unexpectedly still running
```
The above error would tell us that the the ncat command was run  against the INTA router and the echo server was also connected to the INTA router. That would help you quickly know which router logs to look at.

--- a/tests/system_tests_tcp_adaptor.py
+++ b/tests/system_tests_tcp_adaptor.py
@@ -1040,6 +1040,7 @@ class CommonTcpTests:
     def run_ncat(self,
                  port,
                  logger,
+                 name="run_ncat",
                  expect=Process.EXIT_OK,
                  timeout=10,
                  data=b'abcd',
@@ -1055,12 +1056,12 @@ class CommonTcpTests:
                 ncat_cmd.append('--ssl-key')
                 ncat_cmd.append(CLIENT_PRIVATE_KEY_NO_PASS)
         if len(data) > 4:
-            logger.log(f"Starting ncat {ncat_cmd} and large input len={len(data)}")
+            logger.log(f"Starting ncat {ncat_cmd} and large input len={len(data)}, name={name}")
         else:
-            logger.log(f"Starting ncat {ncat_cmd} and input {data}")
+            logger.log(f"Starting ncat {ncat_cmd} and input {data}, name={name}")
         p = self.popen(
             ncat_cmd,
-            stdin=PIPE, stdout=PIPE, stderr=PIPE, expect=expect)
+            stdin=PIPE, stdout=PIPE, stderr=PIPE, expect=expect, name=name)
         out, err = p.communicate(input=data, timeout=timeout)
         try:
             p.teardown()
@@ -1085,7 +1086,8 @@ class CommonTcpTests:
         out = self.run_ncat(ncat_port, logger,
                             data=large_msg if use_large_msg else data,
                             use_ssl=use_ssl,
-                            use_client_cert=use_client_cert)
+                            use_client_cert=use_client_cert,
+                            name=name)
         if use_large_msg:
             logger.log(f"run_ncat large_msg returns length: {len(out)}")
         else:

